clients/src/test/java/org/apache/kafka/common/network/SelectorTest.java

```diff
@@ -72,6 +74,9 @@
  * A set of tests for the selector. These use a test harness that runs a simple socket server that echos back responses.
  */
 public class SelectorTest {
+    @Rule
+    final public Timeout globalTimeout = Timeout.millis(60000);
+
```

Comment:
By the way, I think it's very important to have timeouts like this on JUnit tests.  In this case, without the timeout it just hangs many times on failure (locally at least)